src/rule-creator.ts

```diff
@@ -0,0 +1,51 @@
+import type { TSESLint } from '@typescript-eslint/experimental-utils'
+import { ESLintUtils } from '@typescript-eslint/experimental-utils'
+import type { NGRX_MODULE } from './utils'
+import { docsUrl, MODULE_PATHS, ngrxVersionSatisfies } from './utils'
+
+export type NgRxRuleModule<TMessageIds extends string = string> =
+  TSESLint.RuleModule<TMessageIds, unknown[]> & {
+    meta: { module: NGRX_MODULE; version?: string }
```

Comment:
1. I'd prefer not to call this `module`, as `module` already has meaning in the JS world and can be problematic in case of destructuring, for example.

Some suggestions are: `moduleName`, `ngrxModule` or `ngrxModuleName`.

2. Why is it required? I'm asking because I was imagining that it would be great if we changed all the rules to use this _new way_ (of course, the change would be in another PR), thus keeping all the rules using a single implementation.

Reply:
1. You're right, I'm changing it to `ngrxModule `
2. It's currently set in one of the config scripts, and it's set based on the folder of the rule. I have kept it this way in order to change too much. In the future we can use `createRule` for all of our rules, in those case I would expect that `ngrxModule` is required because we rely on it to build specific configs. After all rules are "refactored", we can remove the code that sets the module atomically based on the folder.